brion/csvConfig.cpp

```diff
@@ -73,43 +52,43 @@ struct CsvConfig::Impl
         if (file.fail())
             throw std::runtime_error("Could not open file `" + uri + "`");
 
-        std::string contents;
-
-        file.seekg(0, std::ios::end);
-        contents.reserve(file.tellg());
-        file.seekg(0, std::ios::beg);
+        std::vector<std::string> lines;
 
-        contents.assign((std::istreambuf_iterator<char>(file)),
-                        std::istreambuf_iterator<char>());
+        for (std::string line; std::getline(file, line);)
+            lines.push_back(line);
 
-        const auto lines = explode(contents, '\n');
         for (const auto& line : lines)
-            table.push_back(explode(line, ' '));
-
-        { // Verify layout
-            const size_t numColumns = table.front().size();
-            for (const auto& row : table)
-            {
-                if (numColumns != row.size())
-                    throw std::runtime_error(
-                        "Number of columns inconsistent in `" + uri + "`");
-            }
+            table.push_back(explode_whitespace(line));
```

Comment:
You can merge the two loops and get rid of the line vector.

Reply:
Ah yes, I am blind :)